jest.config.json

```diff
@@ -0,0 +1,4 @@
+{
+    "testEnvironment": "jsdom",
```

Comment:
I don't think we should be using jsdom.
We should just run the test in a web browser.

Reply:
Unfortunately Jest can't run in a web browser on its own. There are some options we can consider here (but please let me know if there's any other options I may not be considering):

1. Use something like Jest + JSDOM for small, focused unit tests (like this PR) and then work on to include larger end-to-end integration tests that will run in a web browser.
2. Stick to Jest for small, focused unit tests but remove JSDOM as a test environment. Any tests that need to be run in a browser can be included in our integration testing setup
3. Run all unit tests in the browser (e.g. with something like [`jest-puppeteer`](https://jestjs.io/docs/puppeteer)) regardless of whether we decide to include end-to-end integration tests in the near future
4. Stick to writing larger end-to-end integration tests in a web browser and don't include any small, focused unit tests whatsoever

_Option 1_ was my current understanding after our discussions hence why I decided to use JSDOM for simplicity here (plus the developer experience of Jest is pretty nice especially if we want to include unit testing as step in a CI setup). I personally think that's the best option but would love to hear everyone's thoughts. 

I would be open to considering _Option 3_ but we would likely have to switch to another testing tool (Selenium? Karma?) to ensure that we're not only running in Chromium, but I worry that this might be overkill for all the tiny unit tests that we write **and** it might blur the lines between how we differentiate unit tests and larger end-to-end tests.

Reply:
We should do 3. [mocha.js](https://mochajs.org) works in a browser, and we've [used it](https://github.com/WebKit/WebKit/blob/main/Websites/perf.webkit.org/browser-tests/index.html#L7) in another project in WebKit.

Reply:
I ported over all the tests to Mocha in #40. If we all agree on 3, than we can close this PR.